crates/storage/errors/src/db.rs

```diff
@@ -97,18 +98,40 @@ pub struct DatabaseWriteError {
     pub key: Vec<u8>,
 }
 
+impl fmt::Display for DatabaseWriteError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+           f, 
+           "write operation {} failed for key \"{}\" in table {}: {}", 
+           self.operation, 
+           reth_primitives::hex::encode(&self.key), 
+           self.table_name, 
+           self.info
+        )
+    }
+}
```

Comment:
should be possible to derive this too, check out example https://docs.rs/derive_more/latest/derive_more/derive.Display.html#custom-trait-bounds